### PR TITLE
Fix sonar code coverage

### PR DESCRIPTION
**Pull request by willynganga**

### CODE REVIEW

File: flexmark-coverage/pom.xml [hunk 1]
```diff
@@ -0,0 +1,303 @@
...
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.vladsch.flexmark</groupId>
+            <artifactId>flexmark-util-format</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>com.vladsch.flexmark</groupId>
+            <artifactId>flexmark-util-html</artifactId>
```
Comment (Thilo-Acrolinx) [inline flexmark-coverage/pom.xml:222]: Can we omit all transitive dependencies, to make this list shorter and easier to read?
For example, `flexmark-util-format` depends on `flexmark-util-html`, so flexmark-util-html could be omitted?

Comment (willynganga) [reply flexmark-coverage/pom.xml:222]: The purpose of this list of dependencies is to aggregate the Jacoco reports, removing it from the list affects the coverage.

File: pom.xml [hunk 2]
```diff
@@ -259,6 +274,11 @@
                                 <goals>
                                     <goal>report</goal>
                                 </goals>
+                                <configuration>
+                                    <formats>
+                                        <format>XML</format>
```
[the remark this answers is not in the document]
Comment (willynganga) [reply pom.xml:279]: The default one generates the site with many files and we don't even use them. I have no strong preference, so I'll remove it.

File: flexmark-coverage/pom.xml [hunk 3]
```diff
@@ -0,0 +1,303 @@
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
```
[the remark this answers is not in the document]
Comment (willynganga) [reply flexmark-coverage/pom.xml:2]: I used the same as the parent POM in this project. Changing it would mean we change all the others.

**Pull request closed without merging**